src/integration_tests/README.md

```diff
@@ -0,0 +1,71 @@
+# Integration Tests
+
+The Integration Tests package defines Polaris' software verification suite. With it, we can specify testplans that
+encompass the entire software project to verify the integration of our various software projects.
+
+## Defining Tests
+
+Tests are defined using `.yaml` files found under the `testplan/` directory.
+Each testplan should implement the following template:
+
+```yaml
+timeout_sec: # Integer (Optional): Number of seconds allowed for the test to run (default 3 seconds)
+
+required_packages: # List of packages and configuration files
+  - name: # String: one of (boat_simulator, controller, local_pathfinding, network_systems, virtual_iridium, website)
+    configs: # List[String] (Optional): config files relative to the package's config/ folder
+
+inputs: # List of inputs to drive the test
+  - type: # String: one of (ROS, HTTP<TODO>)
+    name: # String: Name of ROS topic or HTTP<TODO> target
+    data: # Varies: Define the type and value of the input
+      dtype: # One of (HTTP type<TODO>, custom_interfaces type, std ROS message type)
+      field_name: # dtype: field_val (both field_name and field_val depend on the dtype)
+                  # if dtype is a builtin type (ex. int32), then set "field_name: <field_val>" as "val: <int val>"
+
+expected_outputs: # List of expected outputs
+  - type: # String: one of (ROS, HTTP<TODO>)
+    name: # String: Name of ROS topic or HTTP<TODO> target
+    data: # Varies: Define the type and value of the output
+      dtype: # One of (HTTP type<TODO>, custom_interfaces type, std ROS message type)
+      DONT_CARE: # Boolean (Optional): Specify True if output data does not matter. Default is False.
+      field_name: # dtype (Optional): field_val (both field_name and field_val depend on the dtype)
+                  # Required if DONT_CARE is False
+```
+
+See [example.yaml](testplans/example.yaml) for an example implementation.
+
+## Invoking with ros2 run
+
+To directly run a testplan, we can use a `ros2 run` command as follows:
+
+```shell
+ros2 run integration_tests run --ros-args -p testplan:=<path/to/testplan.yaml>
+```
+
+For example:
+
+```shell
+ros2 run integration_tests run --ros-args -p testplan:=testplans/example.yaml
+```
+
+## Invoking with ros2 launch
+
+Since the custom testplan `.yaml` files are not valid ROS parameter config files, using them with `ros2 launch` requires
```

Comment:
Could this be done dynamically? Do what this script does when launching the package but don't write into file and pass parameters directly

Reply:
I'm a bit confused about what you're getting at with this. 

If you're talking about the `generate_configs.sh` script, I tried looking into running it during a build command, but colcon doesn't support that. You can do it with CMake, but that's a bit overkill for a Python package.

If you're trying to say "why don't  we pass the custom testplan file and just 'know' that it's the custom file?", then yeah that's doable, but risks confusing the file formats. I put them in a `testplans/` folder to explicitly show that it's a different format.

Reply:
Just for a common and consistent launch interface.

Reply:
I think we can remove, adds unneeded complexity

Reply:
Damn, there was another reason. The global launch code is what sets the log directory. I'll have to set it inside the node.